src/domains/Feedback/ReportHistory/ReportHistory.tsx

```diff
@@ -155,6 +156,38 @@ export default function ReportHistory() {
         refetchReports: () => sendRequest(),
     };
 
+    // Code for filters starts here
+
+    type Filter = FilterFunc<Report>;
+    const filterMap: Record<string, Filter> = {
+        Today: (data) => data.filter(({ date }) => isToday(new Date(date))),
+
+        'This week': (data) => data.filter(({ date }) => isThisWeek(new Date(date))),
+
+        'This month': (data) => data.filter(({ date }) => isThisMonth(new Date(date))),
+
+        'This year': (data) => data.filter(({ date }) => isThisYear(new Date(date))),
+    };
+
+    const search = (searchText: string, data: Report[]) => {
+        const descriptionAccessor = (d: Report) => d.description;
+        return utilSearch(searchText, data, [descriptionAccessor]);
+    };
+
+    const [filters, setFilters] = React.useState<Filter[]>([(reportList: Report[]) => reportList]);
+
+    const handleSearch = React.useCallback(
+        (text: string) =>
+            setFilters(([, ...otherFilters]) => [(filteredList) => search(text, filteredList), ...otherFilters]),
+        [setFilters]
+    );
+
+    const filteredReports = React.useMemo(() => applyFilters(reports || [], filters), [reports, filters]);
+
+    const handleFilterChange = (newFilters: Filter[]) => {
+        setFilters(([searchFilter]) => [searchFilter, ...newFilters]);
+    };
+
```

Comment:
I may have forgotten to point this out, but in the storybook there's an example with the `useFilters` hook here: https://github.com/c2c-project/prytaneum-client/blob/dev/src/components/ListFilter/ListFilter.stories.tsx#L95

I recommend using that.  It will reduce the lines of code here.

Reply:
done